src/components/Reports/SalaryCalculator/MaxAllowableSection/MaxAllowableSection.tsx

```diff
@@ -0,0 +1,176 @@
+import React, { useMemo } from 'react';
+import {
+  Alert,
+  CardContent,
+  CardHeader,
+  Stack,
+  Table,
+  TableBody,
+  TableCell,
+  TableHead,
+  TableRow,
+  Typography,
+} from '@mui/material';
+import { Trans, useTranslation } from 'react-i18next';
+import * as yup from 'yup';
+import { useGoalCalculatorConstants } from 'src/hooks/useGoalCalculatorConstants';
+import { useLocale } from 'src/hooks/useLocale';
+import { currencyFormat } from 'src/lib/intlFormat';
+import { amount } from 'src/lib/yupHelpers';
+import { AutosaveTextField } from '../Autosave/AutosaveTextField';
+import { useSalaryCalculator } from '../SalaryCalculatorContext/SalaryCalculatorContext';
+import {
+  FormattedTableCell,
+  StepCard,
+  StepTableHead,
+} from '../Shared/StepCard';
+
+export interface MaxAllowableStepProps {
+  maxSalary?: number;
+  spouseMaxSalary?: number;
+}
+
+export const MaxAllowableStep: React.FC<MaxAllowableStepProps> = ({
+  // TODO: Get the maximum values for this salary calculation from the MPDX API once they exist
+  maxSalary = 58602,
+  spouseMaxSalary = 48602,
+}) => {
+  const { t } = useTranslation();
+  const locale = useLocale();
+  const { calculation, hcm } = useSalaryCalculator();
+  const { goalMiscConstants: miscConstants } = useGoalCalculatorConstants();
+
+  const schema = useMemo(
+    () =>
+      yup.object({
+        salaryCap: amount(t('Maximum Allowable Salary'), t),
+        spouseSalaryCap: amount(t('Spouse Maximum Allowable Salary'), t),
+      }),
+    [t],
+  );
+
+  const formatCap = (cap: number | null | undefined) => {
+    if (typeof cap !== 'number') {
+      return '-';
+    }
+
+    return currencyFormat(cap, 'USD', locale, {
+      fractionDigits: 0,
+    });
+  };
+
+  const singleCap =
+    miscConstants?.BOARD_APPROVED_SALARY_CALC?.SINGLE_OTHER?.fee;
+  const marriedCap =
+    miscConstants?.BOARD_APPROVED_SALARY_CALC?.MARRIED_OTHER?.fee;
+
+  const [self, spouse] = hcm ?? [];
+  const combinedMaxSalary = maxSalary + spouseMaxSalary;
+  const cap =
+    self?.exceptionSalaryCap.amount ?? (spouse ? marriedCap : singleCap);
+  const overCap = typeof cap === 'number' ? combinedMaxSalary > cap : false;
```

Comment:
@dr-bizz Does this look right? (Will can review the rest of the PR, I just wanted you to look at this part of the business logic).

* I'm reusing the goal calculator constants for the single and married cap (is that right?)
* If they have an `exceptionSalaryCap` from HCM, it overrides the board-approved cap because they have an exception (is that right?)